elastic/integrations
# [elasticsearch] Align mappings with metricbeat

### Summary

Closes https://github.com/elastic/integrations/issues/3914

Align the elasticsearch data streams' mappings with the [metricbeat ones](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/resources/monitoring-es-mb.json). Methodology described in [this issue](https://github.com/elastic/integrations/issues/3914)

### Testing
- build the package `elastic-package build`
- start a 8.5.0 stack `elastic-package stack up -v --version 8.5.0-SNAPSHOT`
- include the package to the existing policy pulled by the running Agent
  - navigate to https://localhost:5601/app/integrations/detail/elasticsearch-0.3.0/overview
  - click "Add elasticsearch"
  - disable "Collect Elasticsearch logs"
  - edit "Collect Elasticsearch metrics" defaults to
 
<img width="699" alt="Screenshot 2022-08-01 at 22 38 47" src="https://user-images.githubusercontent.com/5239883/182242961-b9454f82-0343-4609-b0ed-fb5cbeea4688.png">

  - add the integration to the existing "Elastic-Agent" policy
- open Stack Monitoring and verify that every views load as expected (besides the ones that load Logs, these will be handled in https://github.com/elastic/kibana/issues/137112)

### Known issues
- Cgroup CPU Utilization data does not show up by default and we need to update the es service in the [elastic-package/docker-compose.yml](https://github.com/elastic/elastic-package/blob/main/internal/profile/_static/docker-compose-stack.yml#L3) with cpu limits. Add the following:
```
deploy:
    resources:
        limits:
            cpus: 1.0
```

- I/O Operations Rate is not populated - this is because we're missing the io_stats values from the node_stats API. some infos in this [slack thread](https://elastic.slack.com/archives/C0D8ST60Y/p1659482954487379)

<img width="373" alt="Screenshot 2022-08-01 at 22 43 41" src="https://user-images.githubusercontent.com/5239883/182243648-464cac3a-a48e-4ce1-b92c-f0da7d07b8dc.png">
